test/apps/iodemo/io_demo.cc

```diff
@@ -1007,25 +1073,76 @@ class DemoServer : public P2pDemoCommon {
     }
 
 private:
-    void save_prev_state() {
-        _prev_state = _curr_state;
+    template<io_op_t op_type> static bool
+    get_min_max(const conn_stat_map_t::iterator& i,
+                conn_stat_map_t::iterator& min,
+                conn_stat_map_t::iterator& max)
+    {
+        long i_bytes  = i->second.get_bytes<op_type>();
+
+        if (i_bytes <= min->second.get_bytes<op_type>()) {
+            min = i;
+            return true;
+        } else if (i_bytes >= max->second.get_bytes<op_type>()) {
+            max = i;
+            return true;
+        } else {
+            return false;
+        }
     }
 
-    void report_state() {
-        LOG << "read:" << _curr_state.read_count -
-                          _prev_state.read_count << " ops, "
-            << "write:" << _curr_state.write_count -
-                           _prev_state.write_count << " ops, "
-            << "connections:" << _curr_state.active_conns << "/"
+    void report_state(double time_interval) {
```

Comment:
is it a big change to make conn stats common for client and server?

Reply:
yes, need to unify data structures, client has a lot of code for server_info array and its indexing